apps/SageAccountWeb/src/hooks/useMaybeForceEnable2FA.test.ts

```diff
@@ -0,0 +1,133 @@
+import { renderHook, waitFor } from '@testing-library/react'
+import {
+  ApplicationSessionContextType,
+  useApplicationSessionContext,
+} from 'synapse-react-client/utils/AppUtils/session/ApplicationSessionContext'
+import { vi } from 'vitest'
+import { useLocation, useNavigate, Location } from 'react-router'
+import useMaybeForceEnable2FA from './useMaybeForceEnable2FA'
+
+vi.mock(
+  'synapse-react-client/utils/AppUtils/session/ApplicationSessionContext',
+  () => {
+    return {
+      useApplicationSessionContext: vi.fn(),
+    }
+  },
+)
+
+vi.mock('react-router', () => {
+  return {
+    useLocation: vi.fn(),
+    useNavigate: vi.fn(),
+  }
+})
+
+const mockApplicationSessionContext: ApplicationSessionContextType = {
+  hasInitializedSession: true,
+  refreshSession: vi.fn(),
+  clearSession: vi.fn(),
+  isLoadingSSO: false,
+  token: 'mockToken',
+  twoFactorStatus: {
+    status: 'ENABLED',
+  },
+}
+
+const mockLocation: Location = {
+  key: '',
+  state: undefined,
+  pathname: '/',
+  search: '',
+  hash: '',
+}
+const mockUseApplicationSessionContext = vi.mocked(useApplicationSessionContext)
+const mockNavigate = vi.fn()
+vi.mocked(useNavigate).mockReturnValue(mockNavigate)
+vi.mocked(useLocation).mockReturnValue(mockLocation)
+
+describe('useMaybeForceEnable2FA', () => {
+  beforeEach(() => {
+    vi.clearAllMocks()
+  })
+
+  it('does not redirect and updates state to indicate no redirect when 2FA is enabled', async () => {
+    mockUseApplicationSessionContext.mockReturnValue({
+      ...mockApplicationSessionContext,
+    })
+
+    const hook = renderHook(() => useMaybeForceEnable2FA())
+    await waitFor(() => {
+      expect(hook.result.current.mayForceEnable2FA).toBe(false)
+      expect(mockNavigate).not.toHaveBeenCalled()
+    })
+  })
+  it('does not redirect when user is not logged in', async () => {
+    mockUseApplicationSessionContext.mockReturnValue({
+      ...mockApplicationSessionContext,
+      token: undefined,
+      twoFactorStatus: undefined,
+    })
+
+    const hook = renderHook(() => useMaybeForceEnable2FA())
+    await waitFor(() => {
+      expect(hook.result.current.mayForceEnable2FA).toBe(true)
```

Comment:
We should not redirect back to the source app if not logged in, so I think this "true" response is ok